node/index.ts

```diff
@@ -782,10 +803,37 @@ export class SenderCertificate {
 export class SenderKeyDistributionMessage {
   private readonly nativeHandle: SignalClient.SenderKeyDistributionMessage;
 
+  _unsafeGetNativeHandle(): SignalClient.SenderKeyDistributionMessage {
+    return this.nativeHandle;
+  }
+
   private constructor(nativeHandle: SignalClient.SenderKeyDistributionMessage) {
     this.nativeHandle = nativeHandle;
   }
 
+  static async create(
+    name: SenderKeyName,
+    store: SenderKeyStore
+  ): Promise<SenderKeyDistributionMessage> {
```

Comment:
I think we probably *should* hook up the context parameter, even if we make it [optional](https://www.typescriptlang.org/docs/handbook/functions.html#optional-and-default-parameters), because that'll be groundwork for higher-level transactions like iOS uses.

Reply:
Makes sense. I'm struggling to see how to approach it though. It feels like `Managed::to_raw` (https://docs.rs/neon/0.7.1/src/neon/handle/mod.rs.html#25) is the right way to go from a `Handle<JsValue>` to a `*mut c_void` which we can stick into the libsignal `Context`. However `Managed::from_raw` requires an `Env` (though it is not actually used for `JsValue`'s from_raw) and I can't find any public API in Neon that returns an `Env`.

Maybe thinking about this wrong?

Would propose since we don't need ctx right now, add it later as an optional when we have everything else sorted.

Reply:
Hm, we're going to need to keep it alive somehow, which makes me think we'll need a [Root](https://docs.rs/neon/0.7.1-napi/neon/handle/struct.Root.html) somewhere. We could then do the Swift/C thing and have our low-level Context actually be a reference to the Root.

```rust
let mut root = object.root(&mut cx);
let opaque_context = &mut root as *mut c_void;
```

It's not Rusty at all, though. An alternate approach would be to change the Context type to be `Box<dyn Any>` or `&dyn Any`, which can at least be downcast back safely.

But all right, we can add it later since it's not straightforward.